lib/ngx/re.lua

```diff
@@ -0,0 +1,158 @@
+-- I hereby assign copyright in this code to the lua-resty-core project,
+-- to be licensed under the same terms as the rest of the code.
+
+local ffi = require 'ffi'
+local bit = require "bit"
+local base = require "resty.core.base"
+local core_regex = require "resty.core.regex"
+
+
+local C = ffi.C
+local sub = string.sub
+local type = type
+local band = bit.band
+local new_tab = base.new_tab
+local tostring = tostring
+local tonumber = tonumber
+local math_max = math.max
+local re_match_compile = core_regex.re_match_compile
+local destroy_compiled_regex = core_regex.destroy_compiled_regex
+
+
+local FLAG_DFA           = 0x02
+
+
+local PCRE_ERROR_NOMATCH = -1
+
+
+local split_ctx = new_tab(0, 1)
+local max_split_res_arr_n = 4
```

Comment:
BTW, constant names should be in all upper case, like `PCRE_ERROR_NOMATCH` and `FLAG_DFA` above.